model/project_vars.go

```diff
@@ -212,19 +203,19 @@ func (projectVars *ProjectVars) FindAndModify(varsToDelete []string) (*adb.Chang
 	)
 }
 
-func (projectVars *ProjectVars) GetRestrictedVars() map[string]string {
-	restrictedVars := map[string]string{}
-
+func (projectVars *ProjectVars) GetVars(t *task.Task) map[string]string {
+	vars := map[string]string{}
 	for k, v := range projectVars.Vars {
-		if projectVars.RestrictedVars[k] {
-			restrictedVars[k] = v
+		if !projectVars.AdminOnlyVars[k] {
+			vars[k] = v
 		}
 	}
-	return restrictedVars
+	for k, v := range projectVars.GetAdminOnlyVars(t) {
+		vars[k] = v
+	}
+	return vars
 }
 
-// GetAdminOnlyVars currently returns admin only variables for mainline commits and project admins.
-// Will change to GetVars on EVG-16045 after removing restricted vars.
```

Comment:
I think rather than GetVars calling this function and looping through all vars twice, this could be a helper function that just determines the "shouldUseAdminOnlyVars" part, and then GetVars can just loop through the vars once and include private vars if shouldUseAdminOnlyVars is true.

Reply:
done